processor/src/test/java/org/mapstruct/ap/test/conditional/basic/ConditionalMappingTest.java

```diff
@@ -119,6 +122,218 @@ public void conditionalMethodInUsesContextMapper() {
+
+        TargetPropertyNameModel.Employee employee = new TargetPropertyNameModel.Employee();
+        mapper.map( employeeDto, employee, utils );
+        assertThat( employee.getLastName() ).isNull();
+        assertThat( employee.getFirstName() ).isNull();
+        assertThat( Set.of(
+                "firstName",
+                "lastName",
+                "title",
+                "country" )
+        ).isEqualTo( utils.visited );
+        assertThat( Set.of( "EmployeeDto" ) ).isEqualTo( utils.visitedSources );
+        assertThat( Set.of( "Employee" ) ).isEqualTo( utils.visitedTargets );
+    }
+
+    @ProcessorTest
+    @WithClasses({
+            ConditionalMethodInMapperWithAllExceptTarget.class
+    })
+    public void conditionalMethodInMapperWithAllExceptTarget() {
+        ConditionalMethodInMapperWithAllExceptTarget mapper
+                = ConditionalMethodInMapperWithAllExceptTarget.INSTANCE;
+
+        ConditionalMethodInMapperWithAllExceptTarget.PresenceUtils utils =
+                new ConditionalMethodInMapperWithAllExceptTarget.PresenceUtils();
+
+        TargetPropertyNameModel.EmployeeDto employeeDto = new TargetPropertyNameModel.EmployeeDto();
+        employeeDto.setFirstName( "  " );
+        employeeDto.setLastName( "Testirovich" );
+        employeeDto.setCountry( "US" );
+        employeeDto.setAddresses( List.of( new TargetPropertyNameModel.AddressDto( "Testing St. 6" ) ) );
+
+        TargetPropertyNameModel.Employee employee = mapper.map( employeeDto, utils );
+        assertThat( employee.getLastName() ).isEqualTo( "Testirovich" );
+        assertThat( employee.getFirstName() ).isEqualTo( "  " );
+        assertThat( Set.of(
+                "firstName",
+                "lastName",
+                "title",
+                "country",
+                "street" )
+        ).isEqualTo( utils.visited );
+        assertThat( Set.of( "EmployeeDto", "AddressDto" ) )
+                .isEqualTo( utils.visitedSources );
+    }
+
+    @ProcessorTest
+    @WithClasses({
+        ConditionalMethodWithTargetPropertyNameInContextMapper.class
+    })
+    public void conditionalMethodWithTargetPropertyNameInUsesContextMapper() {
+      ConditionalMethodWithTargetPropertyNameInContextMapper mapper
+        = ConditionalMethodWithTargetPropertyNameInContextMapper.INSTANCE;
+
+      ConditionalMethodWithTargetPropertyNameInContextMapper.PresenceUtils utils =
+        new ConditionalMethodWithTargetPropertyNameInContextMapper.PresenceUtils();
+
+      TargetPropertyNameModel.EmployeeDto employeeDto = new TargetPropertyNameModel.EmployeeDto();
+      employeeDto.setLastName( "  " );
+      employeeDto.setCountry( "US" );
+      employeeDto.setAddresses( List.of( new TargetPropertyNameModel.AddressDto( "Testing St. 6" ) ) );
+
+      TargetPropertyNameModel.Employee employee = mapper.map( employeeDto, utils );
+      assertThat( employee.getLastName() ).isNull();
```

Comment:
shouldn't you assert lastname and firstname as well?

Reply:
Will fix tomorrow